src/lib.rs

```diff
@@ -1049,6 +1049,17 @@ impl Connection {
         Ok(done)
     }
 
+    /// Return idle timeout value
+    fn get_idle_timeout(&mut self) -> time::Duration {
```

Comment:
```suggestion
    fn idle_timeout(&mut self) -> time::Duration {
```

Reply:
Also, could you please move this method at the end of the `impl` block? That is, after `should_update_max_data()` (or above it).

Just so people reading from top to bottom get to the more meatier methods first, and the smaller helper methods don't get in the way.